lang/java/avro/src/main/java/org/apache/avro/specific/SpecificDatumReader.java

```diff
@@ -128,31 +150,43 @@ private boolean trustAllPackages() {
     return (trustedPackages.size() == 1 && "*".equals(trustedPackages.get(0)));
   }
 
-  private void checkSecurity(Class clazz) throws ClassNotFoundException {
-    if (trustAllPackages() || clazz.isPrimitive()) {
+  private void checkSecurity(String className) throws ClassNotFoundException {
+    if (trustAllPackages() || PRIMITIVES.contains(className)) {
       return;
     }
 
+    for (String trustedClass : getTrustedClasses()) {
+      if (className.equals(trustedClass)) {
+        return;
+      }
+    }
+
     boolean found = false;
```

Comment:
nit: This flag is not really needed. Above you use `return`. You could do the same for trustedPackages too.

Reply:
That's right, it's "old code keep". That's for pointing out, let me clean this.